### PR TITLE
Update file name parsing for phase 3, and refactor lint

### PR DESCRIPTION
**Pull request by dougfelt**
Adds noto_names.py to hold methods related to noto font naming and name table entries, and refactors noto_lint to use it.

### CODE REVIEW

**Comment by dougfelt** [comment]
@jamesgk sorry for the large commits, nothing too hairy is going on though.

noto_lint now depends on a data file that is generated by the noto_names tool, but I didn't check that in or add a warning for it in lint.  Still have to figure out whether the phase should affect this data directly, or just be applied in lint.

File: nototools/noto_lint.py [hunk 1]
```diff
@@ -736,6 +603,47 @@ def _check_unused_names():
           warn('name/unused', 'Name', 'Name table has record #%d: "%s"' %
                (i, names[i]), is_error=False)
 
+    def _noto_font_from_font_props(font_props):
+      fields = """
+          filepath,family,style,script,variant,width,weight,slope,fmt,
+          manufacturer,license_type,is_hinted,is_mono,is_UI,is_display,
+          is_cjk,subset
```
Comment (jamesgk) [inline nototools/noto_lint.py]: It seems like this same attribute list is replicated in a few places (here and in the NotoFont and FontProps definitions), could it be defined once somewhere?

File: nototools/noto_names.py [hunk 2]
```diff
@@ -0,0 +1,711 @@
...
+    'LGC': None,
+    'Zsye': None,
+}
+
+
+# copied from noto_lint, we should have a better place for it.
+def preferred_script_name(script_key):
+  try:
+    return unicode_data.human_readable_script_name(script_key)
+  except:
```
Comment (jamesgk) [inline nototools/noto_names.py]: Could this catch a more specific exception? There's another `except:` later in the file also.

**Comment by jamesgk** [comment]
Just minor comments, LGTM

**Comment by jamesgk** [comment]
Oh, something else -- a unit test needs to be updated due to changes to noto_fonts in ede5ab9c790f39b9a146560a311a0773feb1e2f7:

```diff
--- a/nototools/unittests/font_tests.py
+++ b/nototools/unittests/font_tests.py
@@ -178,7 +178,7 @@ class TestNames(FontTest):
         family_names = '%s|%s' % (condensed_name_nosp, name_nosp)
 
         filename_match = noto_fonts.match_filename(filename, family_names)
-        family, _, _, _, _, weight, slope, _ = filename_match.groups()
+        family, _, _, _, _, _, _, _, weight, slope, _ = filename_match.groups()
 
         if family == condensed_name_nosp:
             family = self.condensed_family_name
```

File: nototools/noto_lint.py (hunk 1 above)
Comment (dougfelt) [inline nototools/noto_lint.py]: I'm planning on nuking the FontProps definition and just using NotoFont.  Might rename NotoFont to FontProps, but either way I don't think I need two definitions of nearly the same struct.

**Pull request merged**